AppBuilder/ABFactory.js

```diff
@@ -97,7 +108,195 @@ export default class ABFactory extends ABFactoryCore {
       (Object.keys(platformRules) || []).forEach((k) => {
          this.rules[k] = platformRules[k];
       });
+
+      // Setup a listener for this Object to catch updates from the relay
+      this.network.on("object", async (context, data) => {
+         if (context.error != null) {
+            context.callback?.(context.error);
+            this.AB.analytics.logError(error);
+            return;
+         }
+         const obj = this.datacollections(
+            (datacollection) => datacollection.datasource.id === context.id,
+         )[0]?.datasource;
+         if (obj == null) {
+            context.callback?.(new Error(data));
+            this.AB.analytics.logError(data);
+            return;
+         }
+         if (obj.name != null) {
+            console.log(":: name:", obj.name, {
+               ":: context:": context,
+               ":: data:": data,
+            });
+         } else {
+            console.log(":: context", context, {
+               ":: data": data,
+            });
+         }
+         switch (context.verb) {
+            case "create":
+               // we are being alerted of a NEW object instance.
+               // this might come as a result of our own local().create()
+               // or the server might initiate a push of new data.
+
+               // if data does not already exist locally create it
+               await obj.model().local().syncRemoteMaster(data);
+               // alert any DataCollections that are using this
+               // object that there might be new data for them to
+               // use.
+               obj.emit("CREATE", data);
+               break;
+            case "update":
+               // we are being alerted of UPDATED data from the server.
+
+               // response can be in format:
+               // {
+               //     status:"success",
+               //     data:{ data obj }
+               // }
+               // if this is a response to one of our .updates()
+               if (context.jobID) {
+                  // Where is `latestUpdates` set?
+                  // @see ABModelRelay.js maybe?
+                  if (obj.latestUpdates == null) {
+                     context.callback?.(new Error(data));
+                     this.AB.analytics.logError(data);
+                     return;
+                  }
+                  // if this is the last update we sent for this object
+                  else if (context.jobID !== obj.latestUpdates[data.uuid]) {
+                     context.callback?.(new Error(data));
+                     this.AB.analytics.logError(data);
+                     return;
+                  } else delete obj.latestUpdates[data.uuid];
+               }
+
+               if (data.status && data.status == "success") {
+                  data = data.data || data;
+               }
+               if (data == null) {
```

Comment:
we might want to put this check above where we look at `data.status`

Reply:
https://github.com/digi-serve/appbuilder_platform_mobile/pull/22/commits/d8f46a679fe835edcfcf516ae815f37814be6d81